# 1. Data Analysis

Librerias relevantes para el análisis 

In [97]:

import pandas as pd 
import os
import numpy as np 
from sklearn.linear_model import LinearRegression


1. Importando la base de datos siguiendo las indicaciones

In [98]:
url="https://raw.githubusercontent.com/d2cml-ai/CausalAI-Course/main/data/wage2015_subsample_inference.csv"

In [99]:
wage_gap=pd.read_csv(url)

In [100]:
wage_gap.columns

Index(['rownames', 'wage', 'lwage', 'sex', 'shs', 'hsg', 'scl', 'clg', 'ad',
       'mw', 'so', 'we', 'ne', 'exp1', 'exp2', 'exp3', 'exp4', 'occ', 'occ2',
       'ind', 'ind2'],
      dtype='object')

2. Reporte de NAs

In [101]:
wage_gap.isna().sum()

rownames    0
wage        0
lwage       0
sex         0
shs         0
hsg         0
scl         0
clg         0
ad          0
mw          0
so          0
we          0
ne          0
exp1        0
exp2        0
exp3        0
exp4        0
occ         0
occ2        0
ind         0
ind2        0
dtype: int64

*No existen valores missing en la base de datos*

3. Reporte de estadísticas descriptivas

In [102]:
wage_gap.exp1.describe()

count    5150.000000
mean       13.760583
std        10.609465
min         0.000000
25%         5.000000
50%        10.000000
75%        21.000000
max        47.000000
Name: exp1, dtype: float64

In [103]:
wage_gap.describe()

rownames         wage        lwage          sex          shs  \
count   5150.000000  5150.000000  5150.000000  5150.000000  5150.000000   
mean   15636.346602    23.410410     2.970787     0.444466     0.023301   
std     9700.744395    21.003016     0.570385     0.496955     0.150872   
min       10.000000     3.021978     1.105912     0.000000     0.000000   
25%     7132.500000    13.461538     2.599837     0.000000     0.000000   
50%    15260.000000    19.230769     2.956512     0.000000     0.000000   
75%    24337.750000    27.777778     3.324236     1.000000     0.000000   
max    32643.000000   528.845673     6.270697     1.000000     1.000000   

               hsg          scl          clg           ad           mw  ...  \
count  5150.000000  5150.000000  5150.000000  5150.000000  5150.000000  ...   
mean      0.243883     0.278058     0.317670     0.137087     0.259612  ...   
std       0.429465     0.448086     0.465616     0.343973     0.438464  ...   
min       0.000000     0.000000     0.000000     0.000000     0.000000  ...   
25%       0.000000     0.000000     0.000000     0.000000     0.000000  ...   
50%       0.000000     0.000000     0.000000     0.000000     0.000000  ...   
75%       0.000000     1.000000     1.000000     0.000000     1.000000  ...   
max       1.000000     1.000000     1.000000     1.000000     1.000000  ...   

                we           ne         exp1         exp2         exp3  \
count  5150.000000  5150.000000  5150.000000  5150.000000  5150.000000   
mean      0.216117     0.227767    13.760583     3.018925     8.235867   
std       0.411635     0.419432    10.609465     4.000904    14.488962   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     5.000000     0.250000     0.125000   
50%       0.000000     0.000000    10.000000     1.000000     1.000000   
75%       0.000000     0.000000    21.000000     4.410000     9.261000   
max       1.000000     1.000000    47.000000    22.090000   103.823000   

              exp4            occ         occ2            ind         ind2  
count  5150.000000    5150.000000  5150.000000    5150.000000  5150.000000  
mean     25.118038    5310.737476    11.670874    6629.154951    13.316893  
std      53.530225   11874.356080     6.966684    5333.443992     5.701019  
min       0.000000      10.000000     1.000000     370.000000     2.000000  
25%       0.062500    1740.000000     5.000000    4880.000000     9.000000  
50%       1.000000    4040.000000    13.000000    7370.000000    14.000000  
75%      19.448100    5610.000000    17.000000    8190.000000    18.000000  
max     487.968100  100000.000000    22.000000  100000.000000    22.000000  

[8 rows x 21 columns]

**Interpretación** : se puede extraer información relevante si nos enfocamos en la variable del salario por hora del individuo. Se puede observar que el salario promedio de la muestra es de 2.97 dólares. No obstante, este valor puede verse sesgado por la presencia de valores altos. Por ejemplo, existe una persona que tiene un salario por hora de 6.27 dólares. Para obtener información menos sesgada sobre el salario se recurre a la mediana, la cual nos dice que el 50% de la muestra posee un salario por hora que obscila entre los 1.11 y 2.96 dólares. La desviación estándar es baja , con un valor de 0.57 precisamente por la baja dispersión de los datos debido al proceso de estandarización. 

4. Obtener el número de mujeres con un college degree o por encima que tengan un salario del 25% más rico de la muestra

In [104]:
def salary_class(ing):
    if ((ing >=3.021978) & (ing <= 13.461538)):
        ing_class= "25_less_rich"
    elif ((ing >=27.77778) & (ing <= 528.845673)):
        ing_class= "25_richest"
    else:
        ing_class= "normal"
    return ing_class

In [105]:
wage_gap['wage_classification']=wage_gap['wage'].apply(lambda x: salary_class(x))
wage_gap

rownames       wage     lwage  sex  shs  hsg  scl  clg   ad   mw  ...  \
0           10   9.615385  2.263364  1.0  0.0  0.0  0.0  1.0  0.0  0.0  ...   
1           12  48.076923  3.872802  0.0  0.0  0.0  0.0  1.0  0.0  0.0  ...   
2           15  11.057692  2.403126  0.0  0.0  1.0  0.0  0.0  0.0  0.0  ...   
3           18  13.942308  2.634928  1.0  0.0  0.0  0.0  0.0  1.0  0.0  ...   
4           19  28.846154  3.361977  1.0  0.0  0.0  0.0  1.0  0.0  0.0  ...   
...        ...        ...       ...  ...  ...  ...  ...  ...  ...  ...  ...   
5145     32620  14.769231  2.692546  0.0  0.0  0.0  0.0  1.0  0.0  0.0  ...   
5146     32624  23.076923  3.138833  1.0  0.0  0.0  1.0  0.0  0.0  0.0  ...   
5147     32626  38.461538  3.649659  0.0  0.0  0.0  0.0  0.0  1.0  0.0  ...   
5148     32631  32.967033  3.495508  0.0  0.0  1.0  0.0  0.0  0.0  0.0  ...   
5149     32643  17.307692  2.851151  0.0  0.0  0.0  0.0  0.0  1.0  0.0  ...   

       ne  exp1  exp2    exp3     exp4     occ  occ2     ind  ind2  \
0     1.0   7.0  0.49   0.343   0.2401  3600.0    11  8370.0    18   
1     1.0  31.0  9.61  29.791  92.3521  3050.0    10  5070.0     9   
2     1.0  18.0  3.24   5.832  10.4976  6260.0    19   770.0     4   
3     1.0  25.0  6.25  15.625  39.0625   420.0     1  6990.0    12   
4     1.0  22.0  4.84  10.648  23.4256  2015.0     6  9470.0    22   
...   ...   ...   ...     ...      ...     ...   ...     ...   ...   
5145  0.0   9.0  0.81   0.729   0.6561  4700.0    16  4970.0     9   
5146  0.0  12.0  1.44   1.728   2.0736  4110.0    13  8680.0    20   
5147  0.0  11.0  1.21   1.331   1.4641  1550.0     4  3680.0     6   
5148  0.0  10.0  1.00   1.000   1.0000  2920.0     9  6570.0    11   
5149  0.0  14.0  1.96   2.744   3.8416  1610.0     5  7460.0    14   

      wage_classification  
0            25_less_rich  
1              25_richest  
2            25_less_rich  
3                  normal  
4              25_richest  
...                   ...  
5145               normal  
5146               normal  
5147           25_richest  
5148           25_richest  
5149               normal  

[5150 rows x 22 columns]

In [106]:
wage_gap=wage_gap.assign(case=1)

In [107]:
women_condition=wage_gap['sex']==1
men_condition=wage_gap['sex']==0

In [108]:
wage_only_women=wage_gap[women_condition]
wage_only_men=wage_gap[men_condition]

In [109]:
wage_women_group_a=wage_only_women.groupby(['clg','wage_classification']).agg({'case':['sum']})
wage_women_group_a

case
                         sum
clg wage_classification     
0.0 25_less_rich         484
    25_richest           295
    normal               715
1.0 25_less_rich         107
    25_richest           237
    normal               451

Existen 237 mujeres con un college dregee que poseen un salario del 25% más rico de la muestra. 

In [110]:
wage_only_women.groupby(['ad','wage_classification']).agg({'case':['sum']})

case
                         sum
ad  wage_classification     
0.0 25_less_rich         558
    25_richest           357
    normal               973
1.0 25_less_rich          33
    25_richest           175
    normal               193

Existen 175 mujeres con un advanced degree que poseen un salario del 25% más rico de la muestra. En ambos dataframes, las mujeres que no poseen alguno de estos grados académicos tienden a ser más propensas a tener un salario del 25% menos rico de la muestra, en comparación a las mujeres que sí poseen un mayor nivel académico. Ello nos lleva a inferir que probablemente el nivel educativo puede influir en la obtención de mejores salarios. 

5. 

In [111]:
wage_only_men.groupby(['hsg','wage_classification']).agg({'case':['sum']})

case
                          sum
hsg wage_classification      
0.0 25_less_rich          369
    25_richest            646
    normal               1004
1.0 25_less_rich          309
    25_richest            107
    normal                426

Existen 107 trabajadores hombres que poseen un salario perteneciente al 25% más rico de la muestra. 

In [112]:
wage_only_men.groupby(['shs','wage_classification']).agg({'case':['sum']})

case
                          sum
shs wage_classification      
0.0 25_less_rich          634
    25_richest            747
    normal               1389
1.0 25_less_rich           44
    25_richest              6
    normal                 41

Existen solo 6 trabajadores hombres que tienen un salario perteneciente al 25% más rico de la muestra. Para ambos casos, los hombres que no poseen alguno de estos grados educativos son más propensos a poseer un salario pertenenciente al 25% más pobre. Si comparamos estos resultados con el anterior grupo, podemos observar que hay más mujeres con un salario altísimo en comparación con los hombres, pero ello puede deberse al nivel educativo que poseen. 

6. Creando dos dataframes

In [113]:
wage_gap

rownames       wage     lwage  sex  shs  hsg  scl  clg   ad   mw  ...  \
0           10   9.615385  2.263364  1.0  0.0  0.0  0.0  1.0  0.0  0.0  ...   
1           12  48.076923  3.872802  0.0  0.0  0.0  0.0  1.0  0.0  0.0  ...   
2           15  11.057692  2.403126  0.0  0.0  1.0  0.0  0.0  0.0  0.0  ...   
3           18  13.942308  2.634928  1.0  0.0  0.0  0.0  0.0  1.0  0.0  ...   
4           19  28.846154  3.361977  1.0  0.0  0.0  0.0  1.0  0.0  0.0  ...   
...        ...        ...       ...  ...  ...  ...  ...  ...  ...  ...  ...   
5145     32620  14.769231  2.692546  0.0  0.0  0.0  0.0  1.0  0.0  0.0  ...   
5146     32624  23.076923  3.138833  1.0  0.0  0.0  1.0  0.0  0.0  0.0  ...   
5147     32626  38.461538  3.649659  0.0  0.0  0.0  0.0  0.0  1.0  0.0  ...   
5148     32631  32.967033  3.495508  0.0  0.0  1.0  0.0  0.0  0.0  0.0  ...   
5149     32643  17.307692  2.851151  0.0  0.0  0.0  0.0  0.0  1.0  0.0  ...   

      exp1  exp2    exp3     exp4     occ  occ2     ind  ind2  \
0      7.0  0.49   0.343   0.2401  3600.0    11  8370.0    18   
1     31.0  9.61  29.791  92.3521  3050.0    10  5070.0     9   
2     18.0  3.24   5.832  10.4976  6260.0    19   770.0     4   
3     25.0  6.25  15.625  39.0625   420.0     1  6990.0    12   
4     22.0  4.84  10.648  23.4256  2015.0     6  9470.0    22   
...    ...   ...     ...      ...     ...   ...     ...   ...   
5145   9.0  0.81   0.729   0.6561  4700.0    16  4970.0     9   
5146  12.0  1.44   1.728   2.0736  4110.0    13  8680.0    20   
5147  11.0  1.21   1.331   1.4641  1550.0     4  3680.0     6   
5148  10.0  1.00   1.000   1.0000  2920.0     9  6570.0    11   
5149  14.0  1.96   2.744   3.8416  1610.0     5  7460.0    14   

      wage_classification  case  
0            25_less_rich     1  
1              25_richest     1  
2            25_less_rich     1  
3                  normal     1  
4              25_richest     1  
...                   ...   ...  
5145               normal     1  
5146               normal     1  
5147           25_richest     1  
5148           25_richest     1  
5149               normal     1  

[5150 rows x 23 columns]

In [115]:
wage_only=wage_gap.loc[:,"lwage"]
wage_only

0       2.263364
1       3.872802
2       2.403126
3       2.634928
4       3.361977
          ...   
5145    2.692546
5146    3.138833
5147    3.649659
5148    3.495508
5149    2.851151
Name: lwage, Length: 5150, dtype: float64

In [88]:
wage_rest=wage_gap.drop(columns=["lwage","wage"])
wage_rest

rownames  sex  shs  hsg  scl  clg   ad   mw   so   we  ...  exp1  exp2  \
0           10  1.0  0.0  0.0  0.0  1.0  0.0  0.0  0.0  0.0  ...   7.0  0.49   
1           12  0.0  0.0  0.0  0.0  1.0  0.0  0.0  0.0  0.0  ...  31.0  9.61   
2           15  0.0  0.0  1.0  0.0  0.0  0.0  0.0  0.0  0.0  ...  18.0  3.24   
3           18  1.0  0.0  0.0  0.0  0.0  1.0  0.0  0.0  0.0  ...  25.0  6.25   
4           19  1.0  0.0  0.0  0.0  1.0  0.0  0.0  0.0  0.0  ...  22.0  4.84   
...        ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...   ...   ...   
5145     32620  0.0  0.0  0.0  0.0  1.0  0.0  0.0  0.0  1.0  ...   9.0  0.81   
5146     32624  1.0  0.0  0.0  1.0  0.0  0.0  0.0  0.0  1.0  ...  12.0  1.44   
5147     32626  0.0  0.0  0.0  0.0  0.0  1.0  0.0  0.0  1.0  ...  11.0  1.21   
5148     32631  0.0  0.0  1.0  0.0  0.0  0.0  0.0  0.0  1.0  ...  10.0  1.00   
5149     32643  0.0  0.0  0.0  0.0  0.0  1.0  0.0  0.0  1.0  ...  14.0  1.96   

        exp3     exp4     occ  occ2     ind  ind2  wage_classification case  
0      0.343   0.2401  3600.0    11  8370.0    18         25_less_rich    1  
1     29.791  92.3521  3050.0    10  5070.0     9           25_richest    1  
2      5.832  10.4976  6260.0    19   770.0     4         25_less_rich    1  
3     15.625  39.0625   420.0     1  6990.0    12               normal    1  
4     10.648  23.4256  2015.0     6  9470.0    22           25_richest    1  
...      ...      ...     ...   ...     ...   ...                  ...  ...  
5145   0.729   0.6561  4700.0    16  4970.0     9               normal    1  
5146   1.728   2.0736  4110.0    13  8680.0    20               normal    1  
5147   1.331   1.4641  1550.0     4  3680.0     6           25_richest    1  
5148   1.000   1.0000  2920.0     9  6570.0    11           25_richest    1  
5149   2.744   3.8416  1610.0     5  7460.0    14               normal    1  

[5150 rows x 21 columns]

In [ ]:
7. 

In [116]:
Y=wage_only
Y

0       2.263364
1       3.872802
2       2.403126
3       2.634928
4       3.361977
          ...   
5145    2.692546
5146    3.138833
5147    3.649659
5148    3.495508
5149    2.851151
Name: lwage, Length: 5150, dtype: float64

In [ ]:
wage_rest.